app/controllers/bcourses_controller.rb

```diff
@@ -0,0 +1,25 @@
+class BcoursesController < ApplicationController
+  require 'lms_api'
+
+  def index
+    # For dev,test,staging, they share the same canvas_test_url
+    canvas_url = Rails.application.credentials.dig(Rails.env.to_sym, :canvas, :url)
```

Comment:
If you're going to do this, you should be adding these to the `credentials.yml.enc` as well as the `credentials/prod.yml.enc` files because the prod versions should be different than the default ones.

Also, if you decide to change this, it is prob better to directly access these (eg: `Rails.application.credentials.canvas.url`)  instead of having to dig them since dig loads the entirety of the secrets into the dig scope.  You can just mark this as resolved if you don't want to do this, but just food for thought.

Reply:
Hi Connor, I've made the following two points regarding your suggestions:
- Move the credentials that are only used in prod environment from `credentials.yml.enc` to `credentials/prod.yml.enc`.
- Use the `Rails.application.credentials.canvas_url` since all the dev/test/staging env should use the same URL. This change is also made in the  `credentials/prod.yml.enc`.
- Because of the modification in the controller, I made corresponding changes in `rspec` file.